container_pipeline/cleanup_registry/garbagecollector.py

```diff
@@ -181,88 +134,119 @@ def _gcollect(self):
-
-                    # If index needs to be checked
-                    if index_check:
-                        index_tags_list = self._index_containers.get(
-                            registry_name
-                        )
-                        # If there are no tags in index to match with or
-                        # registry tag not in the index, then its mismatch.
-                        if not index_tags_list or (
-                                index_tags_list and registry_tag not in
-                                index_tags_list):
-                            index_mismatched = True
-
-                    # If we need to check the exception list
-                    if exception_list_check:
-                        for exp in self._gc_exceptions:
-                            if exp.match(image_name):
-                                exception_list_found = True
-                                break
-
-                    # If we need to check match only list
                     if match_only_check:
                         for exp in self._gc_match_only:
                             if exp.match(image_name):
                                 match_only_list_found = True
                                 break
+                    if exception_list_check:
+                        for exp in self._gc_exceptions:
+                            if exp.match(image_name):
+                                exception_list_found = True
+                                break
 
-                    # Do the actual matching now
-                    if index_check and index_mismatched:
-                        if not exception_list_found:
-                            if match_only_check and not \
-                                    match_only_list_found:
-                                continue
-                            self._mismatched[registry_name].append(
-                                registry_tag
-                            )
+                    self._update_mismatched(
+                        k, item, match_only_check,
+                        match_only_list_found, exception_list_check,
+                        exception_list_found
+                    )
 
-                        else:
-                            if match_only_list_found and not \
-                                    exception_list_found:
-                                self._mismatched[registry_name].append(
-                                    registry_tag
-                                )
-                    else:
-                        if match_only_list_found and not exception_list_found:
-                            self._mismatched[registry_name].append(
-                                registry_tag
-                            )
+        else:
+            registry_info = RegistryInfo(
+                registry_host=self._registry_host,
+                registry_port=self._registry_port,
+                registry_secure=self._registry_secure
```

Comment:
what registry_secure stand for, https? AFAIK we are doing this locally so how does it matter?

Reply:
This is just to make it future proof. It has no affect now, but doing it now means i wont have to change in case we decide to go full secure or something.